### PR TITLE
Add project structure and development section to README.md

### PR DESCRIPTION
**Pull request by kazydek**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [ ] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Added the development section to the main `README.md` document in the repository
- Added the project structure to the main README.md document in the repository
- Removed information on acceptance tests from the README.md in the `ui-api-layer` project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+
+4. Run the `git fetch --all` command to fetch all remote branches.
+
+5. Set up the local `master` branch to track the remote master branch from the upstream repository:
+
+```
+git branch -u upstream/master master
+```
+
+6. Install dependencies.
```
Comment (pkosiec) [inline README.md]: I'd convert the numbered list to separate sections (subheaders). Because the only thing that should be done in order, is installing dependencies before doing anything else.

File: README.md [hunk 2]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+
+Go to the main directory of the project in your workspace location and install required dependencies:
+
+```
+$ cd $GOPATH/src/github.com/{github-username}/kyma/components/ui-api-layer
```
Comment (pkosiec) [inline README.md]: nope - `kyma-project/kyma`, not `{github-username}`

File: README.md [hunk 3]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+
+9. Run unit tests.
+
+After you finish, test your changes. Use the `go test ./...` command to run unit tests or follow test instructions specified in the main `README.md` document of the given project.
+
+10. Perform acceptance tests.
+
+Go to the `tests` directory in the `kyma` repository and run acceptance tests for a given components.
+
+11. When the tests complete successfully, submit the pull request and wait for the owners' review.
```
Comment (pkosiec) [inline README.md]: Instead of this point, I'd suggest adding a separate section with `Contribution` title, which will point to `CONTRIBUTING.md`

Comment (derberg) [inline README.md]: we need to explain that first is a Pr with code changes, once it is accepted, there must be another Pr with version update in Kyma charts. This flow must be explained, that contribution has for now 2 stages

File: README.md [hunk 4]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+## Development
+
+Develop on your remote repository forked from the original repository in Go.
+See the example that uses the [`ui-api-layer`](components/ui-api-layer) project located in the `components` directory in the `kyma` repository but applies to any Go project. This set of instructions uses the recommended [`git workflow`](https://github.com/kyma-project/community/blob/master/git-workflow.md).
+
+1. Fork the repository in GitHub.
+2. Clone the fork to your `$GOPATH` workspace. Use this command to create the folder structure and clone the repository under the correct location:
+
+```
+git clone git@github.com:{github-username}/kyma.git $GOPATH/src/github.com/{github-username}/kyma
```
Comment (derberg) [inline README.md]: kyma-project

File: README.md [hunk 5]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+```
+3. Add a Git remote for the original repository:
+
+```
```
Comment (derberg) [inline README.md]: cd $GOPATH/src/github.com/kyma-project/kyma

File: README.md (hunk 2 above)
Comment (derberg) [reply README.md]: actually cd components/ui-api-layer

File: README.md [hunk 6]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
 
 Read the [documentation](docs/README.md) to learn about the product, its technology stack, and components.
 
+### Project structure
+
+The `kyma` repository has the following structure:
+
+```
+  ├── .github                     # Pull request and issue templates             
+  ├── components                  # Source code of the application
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: `Source code of Kyma components` ?

File: README.md [hunk 7]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+The `kyma` repository has the following structure:
+
+```
+  ├── .github                     # Pull request and issue templates             
+  ├── components                  # Source code of the application                                                   
+  ├── docs                        # Documentation
+  ├── installation                # Installation scripts     
+  ├── resources                   # Helm charts and Kubernetes resources for the Kyma installation
+  ├── tests                       # Acceptance tests
+  ├── tools                       # Source code of utilities used in the installation and testing
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: `Source code of utilities used for the installation, testing, and other purposes `?

File: README.md [hunk 8]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+  ├── tests                       # Acceptance tests
+  ├── tools                       # Source code of utilities used in the installation and testing    
+  ├── CODEOWNERS                  # Repository owners  
+  ├── CODE_OF_CONDUCT.md          # Rules of behavior for maintainers and contributors          
+  ├── CONTRIBUTING.md             # Rules of contribution
+  ├── Jenkinsfile                 # Integration job
+  ├── LICENSE                     # The document with terms for use, reproduction, modification, and distribution of Kyma code
+  ├── NOTICE.md                   # Copyright information
+  ├── README.md                   # The document with the project overview, structure, installation, and development steps
+  ├── azure.Jenkinsfile           # The file that builds and tests the Azure cluster
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: I removed the file structure and descriptions.

File: README.md [hunk 9]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+  ├── ci.Dockerfile               # The file that verifies and tests the Docker image built
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: I removed the file structure and descriptions.

File: README.md [hunk 10]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+  ├── docs                        # Documentation
+  ├── installation                # Installation scripts     
+  ├── resources                   # Helm charts and Kubernetes resources for the Kyma installation
+  ├── tests                       # Acceptance tests
+  ├── tools                       # Source code of utilities used in the installation and testing    
+  ├── CODEOWNERS                  # Repository owners  
+  ├── CODE_OF_CONDUCT.md          # Rules of behavior for maintainers and contributors          
+  ├── CONTRIBUTING.md             # Rules of contribution
+  ├── Jenkinsfile                 # Integration job
+  ├── LICENSE                     # The document with terms for use, reproduction, modification, and distribution of Kyma code
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: I removed the file structure and descriptions.

File: README.md [hunk 11]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+  ├── NOTICE.md                   # Copyright information
+  ├── README.md                   # The document with the project overview, structure, installation, and development steps
+  ├── azure.Jenkinsfile           # The file that builds and tests the Azure cluster
+  ├── ci.Dockerfile               # The file that verifies and tests the Docker image built
+  ├── governance.Jenkinsfile      # The file that validates external links in the documentation
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: I removed the file structure and descriptions.

File: README.md [hunk 12]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+  └── orchestrator.Jenkinsfile    # The file that triggers Kyma integration and runs the CI files of all subprojects based on the introduced changes
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: As above

File: README.md [hunk 13]
```diff
@@ -20,10 +20,85 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
 ## Usage
 
 Kyma comes with the ready-to-use code snippets that you can use to test the extensions and the core functionality. See the list of existing examples in the [`examples`](https://github.com/kyma-project/examples) repository.
+
+## Development
+
+Develop on your remote repository forked from the original repository in Go.
+See the example that uses the [`ui-api-layer`](components/ui-api-layer) project located in the `components` directory in the `kyma` repository but applies to any Go project. This set of instructions uses the recommended [`git workflow`](https://github.com/kyma-project/community/blob/master/git-workflow.md).
+
+1. Fork the repository in GitHub.
```
[the remark this answers is not in the document]
Comment (kazydek) [reply README.md]: Fair enough. I will make a reference to the git-workflow.md in which the process is already described.

File: README.md (hunk 1 above)
Comment (kazydek) [reply README.md]: Aren't subheaders to much for such short instructions/steps that we want to keep simple? We can make bullet points out of the list if you still consider that essential?.

File: README.md (hunk 3 above)
Comment (kazydek) [reply README.md]: Actually, given the comments from @derberg that we need to mention two-PR process, I would keep it this way.

File: README.md [hunk 14]
```diff
@@ -20,10 +20,69 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+Follow these steps:
+
+> **NOTE:** The example assumes you have the `$GOPATH` already set.
+
+1. Fork the repository in GitHub.
+
+2. Clone the fork to your `$GOPATH` workspace. Use this command to create the folder structure and clone the repository under the correct location:
+
+```
+git clone git@github.com:kyma-project/kyma.git $GOPATH/src/github.com/kyma-project/kyma
```
Comment (michal-hudy) [inline README.md]: If instruction is based on forked repository then it should be:
```
git clone git@github.com:{userName}/kyma.git $GOPATH/src/github.com/kyma-project/kyma
```

File: README.md [hunk 15]
```diff
@@ -20,10 +20,62 @@ Living outside of the core product, Kyma allows you to be completely language-ag
 
 Read the [documentation](docs/README.md) to learn about the product, its technology stack, and components.
 
+### Project structure
```
Comment (derberg) [inline README.md]: hmmm, it will annoy me that this is more important than the installation instruction/links

File: README.md [hunk 16]
```diff
@@ -20,10 +20,62 @@ Living outside of the core product, Kyma allows you to be completely language-ag
...
+
+4. Build the project.
+
+Every project runs differently. Follow instructions in the main `README.md` document of the given project to build it.
+
+5. Create a branch and start to develop.
+
+6. Test your changes.
+
+When you finish, run unit and acceptance tests to verify your changes. For the unit tests, follow the instructions specified in the main `README.md` document of the given project. To run the acceptance tests, go to the `tests` directory for details.
```
Comment (derberg) [inline README.md]: my comment was ignored
```
is this really that simple? I don't think so
and what about writing that unit and acceptance tests are required in pull request, and docu updated
```

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -19,3 +19,4 @@ Apart from the general rules described in the `community` repository, every `kym
 * Do not copy charts from the Internet. Customize the Helm charts and simplify them to pertain only to the specific use case. Apply this rule to all documents associated with the charts, such as `README.md` and `NOTES.txt` documents.
 * Follow the `IfNotPresent` pulling policy. Do not use the `latest` tag for all `Deployments` definitions for the local installation.
 * Adjust any data copied from the Internet to the product needs.
+* When you receive the required approvals for your pull request, create another one to update the image version and any configuration changes in relevant Kyma charts.
```
Comment (derberg) [inline CONTRIBUTING.md]: do you really think it is totally clear what is the flow
why 2stage flow?

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -19,4 +19,4 @@ Apart from the general rules described in the `community` repository, every `kym
 * Do not copy charts from the Internet. Customize the Helm charts and simplify them to pertain only to the specific use case. Apply this rule to all documents associated with the charts, such as `README.md` and `NOTES.txt` documents.
 * Follow the `IfNotPresent` pulling policy. Do not use the `latest` tag for all `Deployments` definitions for the local installation.
 * Adjust any data copied from the Internet to the product needs.
-* When you receive the required approvals for your pull request and it is merged, create another one to update the image version and any configuration changes in relevant Kyma charts.
+* When you receive the required approvals for your pull request and it is merged, create another one to update the image version and any configuration changes in relevant Kyma charts. Images are generated and pushed by Continuous Integration system and changes in configuration need to be done manually.
```
Comment (klaudiagrz) [inline CONTRIBUTING.md]: `The Continuous Integration system generates and pushes images. You must implement the configuration changes manually.`

File: README.md [hunk 19]
```diff
@@ -62,9 +62,12 @@ Every project runs differently. Follow instructions in the main `README.md` docu
 
 5. Create a branch and start to develop.
 
-6. Test your changes.
+Do not forget about creating unit and acceptance tests if needed. For the unit tests, follow the instructions specified in the main `README.md` document of the given project. For the acceptance tests, go to the corresponding directory inside `tests` directory for details.
```
Comment (klaudiagrz) [inline README.md]: ``For the details concerning the acceptance tests, go to the corresponding directory inside the `tests` directory.``

File: README.md [hunk 20]
```diff
@@ -62,9 +62,12 @@ Every project runs differently. Follow instructions in the main `README.md` docu
...
+Information how to run changes on cluster without Docker Image can be found in [Develop a service locally without using Docker](docs/kyma/docs/035-gs-local-develop-no-docker.md)
```
Comment (klaudiagrz) [inline README.md]: `Find the information on how to run changes on the cluster without a Docker image in the [Develop a service locally without using Docker](docs/kyma/docs/035-gs-local-develop-no-docker.md) document.`

File: README.md [hunk 21]
```diff
@@ -62,9 +62,12 @@ Every project runs differently. Follow instructions in the main `README.md` docu
...
+
+> For more details about testing go to [Testing Kyma](docs/kyma/docs/026-details-testing.md)
```
Comment (klaudiagrz) [inline README.md]: ``>**NOTE:** For more details about testing, go to the [Testing Kyma](docs/kyma/docs/026-details-testing.md) document.``
